models/things.py

```diff
@@ -241,6 +241,84 @@ def get_by_mqtt_topic(db, topic: T.List[str]):
         return Thing.get_by_type_and_device_id(db, node_type, device_id, vnode_id), None
 
 
+class ShellyPower(Thing):
+    __mapper_args__ = {
+        'polymorphic_identity': 'shelly_power'
+    }
+
+    def get_data_type(self):
+        return DataType.Float
+
+    def get_state_topic(self):
+        return "shellies/{device_id}/emeter/{vnode_id}/power".format(type=self.type, device_id=self.device_id,
+                                                              vnode_id=self.vnode_id)
+
+    def get_action_topic(self):
+        return None
+
+    def process_status(self, db, state, data):
+        LastSeen.update_last_seen(db, self.device_id)
+        state = f"unknown,{state}"
+        print(f"State of {self.name} of type {self.type} is {state}")
+        return super().process_status(db, state, data)
+
+    @classmethod
+    def display_name(cls):
+        return 'Shelly Power'
+
+    @staticmethod
+    def get_mqtt_subscriptions():
+        return 'shellies/+/emeter/+/power',
+
+    @staticmethod
+    def get_by_mqtt_topic(db, topic: T.List[str]):
+        # shellies / node / emeter / vnode / function
+        node_type = 'shelly_power'
+        device_id = topic[1]
+        vnode_id = topic[3]
+        function = topic[4]
+
+        return Thing.get_by_type_and_device_id(db, node_type, device_id, vnode_id), None
+
+
+class ShellyEnergy(Thing):
+    __mapper_args__ = {
+        'polymorphic_identity': 'shelly_energy'
+    }
+
+    def get_data_type(self):
+        return DataType.Float
+
+    def get_state_topic(self):
+        return "shellies/{device_id}/emeter/{vnode_id}/total".format(type=self.type, device_id=self.device_id,
+                                                              vnode_id=self.vnode_id)
+
+    def get_action_topic(self):
+        return None
+
+    def process_status(self, db, state, data):
+        LastSeen.update_last_seen(db, self.device_id)
+        state = f"unknown,{state}"
+        print(f"State of {self.name} of type {self.type} is {state}")
+        return super().process_status(db, state, data)
+
+    @classmethod
+    def display_name(cls):
+        return 'Shelly Energy'
+
+    @staticmethod
+    def get_mqtt_subscriptions():
+        return 'shellies/+/emeter/+/total',
```

Comment:
We could also support Shelly Plug here by returning a list that also includes `"shellies/+/relay/+/energy"` (although that value is in watt-minutes instead of watt-hours the EM and 3EM report I think.